WalletWasabi.Fluent/Controls/CurrencyEntryBox.axaml.cs

```diff
@@ -259,28 +260,28 @@ private bool ValidateEntryText(string preComposedText)
 		return true;
 	}
 
-	protected override void OnKeyDown(KeyEventArgs e)
+	protected override async void OnKeyDown(KeyEventArgs e)
 	{
-		DoPasteCheck(e);
+		await DoPasteCheckAsync(e);
```

Comment:
We need to be absolutely sure that this method won't raise any exception. Otherwise, the application will crash because of this. So:
- If we're absolutely sure that `DoPasteCheckAsync` will never raise an exception, no changes are needed
- Otherwise, we should catch any possible exception with a try-catch block. Eg.

```
private async Task DoPasteCheckAsync(KeyEventArgs e)
{
	try
	{
		var keymap = AvaloniaLocator.Current.GetService<PlatformHotkeyConfiguration>();
		bool Match(IEnumerable<KeyGesture> gestures) => gestures.Any(g => g.Matches(e));

		if (keymap is { } && Match(keymap.Paste))
		{
			await ModifiedPasteAsync();
		}
		else
		{
			base.OnKeyDown(e);
		}
	}
	catch (Exception ex)
	{
		// ...
	}
}
```

Reply:
Examples:

### Done right
```
protected override async void OnKeyDown(KeyEventArgs e)
{
	await HandleIt(e);
}

private Task HandleIt(KeyEventArgs e) 
{
	try 
	{
		throw new ApplicationException("This will be handled gracefully");	
	} 
	catch 
	{
		
	}
}
```

### Done wrong

```
protected override async void OnKeyDown(KeyEventArgs e)
{
	await HandleIt(e);
}

private Task HandleIt(KeyEventArgs e)
{
	throw new ApplicationException("This will be crash the app");	
}
```

Reply:
Thanks, @SuperJMN, I haven't thought about this! I checked the code deeply and in some cases, an exception is thrown. Try catch added.

Reply:
Wonderful @soosr ! Wasabi is more robust thanks to this 😃